api/overdrive.py

```diff
@@ -1037,13 +1037,18 @@ def update_licensepool(self, book_id):
                 exc_info=e
             )
 
-        if status_code != 200:
+        # TODO: If you ask for a book that you know about, and
+        # Overdrive says the book doesn't exist in the collection,
+        # then it's appropriate to update an existing
+        # LicensePool. However we shouldn't be creating a *brand new*
+        # LicensePool for a book Overdrive says isn't in the
+        # collection.
+        if status_code not in (200, 404):
```

Comment:
This is the actual fix for the problem we're seeing: the Overdrive API is telling us that a book is no longer in the collection, but because they're sending a 404 error (rather than a 200 that says "there are no licenses") we're ignoring that information and acting like the book is still present in the collection.